From 13727772c39f7727fb3be94ddd8a1cd46c16a1ee Mon Sep 17 00:00:00 2001
From: Tim Orling <tim.orling@konsulko.com>
Date: Thu, 9 Jun 2022 19:42:31 -0700
Subject: [PATCH] lib/tools-common.in: keystoretool -> luks-srv-app

Swap keystoretool for luks-srv-app, with DISK_ENCRYPTION_CONTEXT
placeholder

Signed-off-by: Tim Orling <tim.orling@konsulko.com>
---
 lib/tools-common.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/tools-common.in b/lib/tools-common.in
index 39f3ca6..2c1b986 100644
--- a/lib/tools-common.in
+++ b/lib/tools-common.in
@@ -736,7 +736,7 @@ initialize_devices() {
     local fakept=`mktemp -q`
     if make_fake_parttable $fakept; then
 	local ppdir=`mktemp -q -d /run/setup-pp.XXXXXX`
-	keystoretool -p > $ppdir/passphrase
+	/usr/sbin/luks-srv-app --get-unique-pass --context-string=@DISK_ENCRYPTION_CONTEXT@ > $ppdir/passphrase
 	if ! create_luks_partitions $ppdir $fakept; then
 	    rm -rf $ppdir
 	    return 1
